### DISCUSSION TITLE
Is this normal?

### DISCUSSION THREAD

**Discussion by jakezp** [Dilbert66/esphome-dsckeybus#81] [General]
Hey!

So I've been playing with ESPHome, a Wemos D1 mini and DSC Powerseries 1864 that is installed in the house we recently moved into. I've managed to the D1 connected to the alarm system and it seems like things are working (sort of). I've got some 'weird things' happening, like some partitions status being unavailable, zones flipping to between whatever state and unavailable, or '00: Unknown' popping up randomly, etc. I've had a look at the logs and it seems quite noisy to me, I'm not sure if it's normal. I've also set up a test panel in Home assistant and monitoring the panel, as well as all the sensors, it looks very busy, states changing constantly (no one else at home opening and closing doors, setting off beams or PIRs, etc. And then the other weirdness is that the alarm would just trigger randomly. We're connected to remote monitoring with armed response, and according to them, they receive a panic activation, which seems very strange.

The wiring are done as per the instructions. The only difference is that I changed the 33k resistors for 15k resistors, since the circuit would not work with 33k resistors. I assume the voltage was too low on yellow and green and I just got garbage CRC errors on the serial monitor (I was still playing around with https://github.com/taligentx/dscKeybusInterface at this stage. After changing the resistors, I actually started getting non-crc error messages.

I'd appreciate it if someone can take a look at my config and logs (attached) and provide some guidance?

Thanks

[logs_DscAlarm_logs.txt](https://github.com/Dilbert66/esphome-dsckeybus/files/11033647/logs_DscAlarm_logs.txt)

**Comment by Dilbert66** [comment]
No, not normal.  You have a misconfigured input somewhere. Check your pinouts. Also, you need to use the 33k if using 10k.  You will destroy the D1 wit the ratio you have since you are applying too much voltage to the pins.  ie 10/15 * 12=8 volts to 3.3 volt rated pins. Not good.    10/33*12=3.6 which is good enough tolerance.   If the circuit didnt work with 33k then you have other issues.

**Reply by jakezp** [reply]
> No, not normal. You have a misconfigured input somewhere. Check your pinouts. Also, you need to use the 33k if using 10k. You will destroy the D1 wit the ratio you have since you are applying too much voltage to the pins. ie 10/15 * 12=8 volts to 3.3 volt rated pins. Not good. 10/33*12=3.6 which is good enough tolerance. If the circuit didnt work with 33k then you have other issues.

@Dilbert66 Thanks for the feedback and confirming that it's not normal. I based the resistors on the voltages I measured:

* AUX- as ground
* AUX+ 13.56v
* Green 8.2V
* Yellow 6.8V

Using https://ohmslawcalculator.com/voltage-divider-calculator, I landed on the following using 15k/10k:
Green: 3.28v
Yellow: 2.72v

Which I assumed should be close enough to 3.3v for D1 and D2.

Using the specified 33k/10k, I would end up with Green: 1.9v, Yellow: 1.58v, which I assumed was too low, hence the CRC errors when monitoring over serial. If I've made an incorrect assumption, or if there is anything else I should check, please let me know.

**Comment by Dilbert66** [comment]
Did you measure with a dc meter on the bus lines? That's not right.  You need an oscilloscope to see the actual peak voltage. Please note that this is a data bus so  what you are measuring is an average of the signal.  The bus voltages are 12 volts peak so you need to use that, not what you measure.

**Reply by jakezp** [reply]
Ok, thanks... So yeah, I measures using the terminals on the board. Thanks for clarifying - I'll remove the resistors with 33k resistors and give it another and see what I get.

I'll report back. 

Thanks for the guidance

**Comment by jakezp** [comment]
@Dilbert66 

So I decided to start from scratch, using ESP32. I redid the wiring exactly per the instructions and I am no longer seeing the 'weird things' like unknown states and things just random flipping between states and unavailable. It definitely seems more stable now. 

I'm however. not able to send any useful commands to the alarm. Things like < and > are working just fine and looking at the ESPHome logs, I do see the commands reach the device (mostly) - see below. But I cannot do things like check alarmed zones (*3), trouble codes (*2), etc. When I try *2 a few times in a row, the alam would arm itself, the attempting to disarm, nothing happens. I've attached my logs again. I have set my log level to VERBOSE while I try to figure all this out.

The following line numbers in the logs show *2 being used to check trouble and then 'D' a few times in an attempt to disarm the system. I tried switching to stay more using 'S', but this also failed. 

Attempted to view system troubles by sending *2 to partition 1:

`[23:05:22][D][Debug:865]: Writing keys: *2 to partition 1, partition disabled: 0 , partition locked: 0`

Partition 1 Armed itself without selecting 'Arm'

`[23:05:23][V][text_sensor:016]: 'P1 line 1': Received new state Armed:`       
`[23:05:23][D][text_sensor:067]: 'P1 line 1': Sending state 'Armed:`
`[23:05:23][V][text_sensor:016]: 'P1 Line 2': Received new state Away`            
`[23:05:23][D][text_sensor:067]: 'P1 Line 2': Sending state 'Away`     

Attempted to Disarm (unsuccessful):

`[23:06:24][D][debug:930]: Setting Alarm state: D to partition 1`
`[23:06:24][D][debug:976]: Disarming ... `

And again:

`[23:06:55][D][debug:930]: Setting Alarm state: D to partition 1`
`[23:06:55][D][debug:976]: Disarming ... `

Again: 

`[23:08:30][D][debug:930]: Setting Alarm state: D to partition 1`
`[23:08:30][D][debug:976]: Disarming ... `

Attempted to switch to Armed Stay:

`[23:09:03][D][debug:930]: Setting Alarm state: S to partition 1`

If you have a moment, do you mind having a look at the logs and providing guidance? What else can I check? Do I still have a wiring issue? Is it something else?

[dsc_logs.txt](https://github.com/Dilbert66/esphome-dsckeybus/files/11105700/dsc_logs.txt)


Thanks

**Comment by Dilbert66** [comment]
I suspect you have a weak signal on your send circuit or something as it looks like the data is being sent correctly but the panel is misreading what it gets.  Your received data is fine.  It's the only thing I can think of honestly as the issue you're having really doesnt make sense otherwise.

**Reply by jakezp** [reply]
Thanks for the feedback. Any suggestions to increase the signal on the send side of things?

**Comment by jakezp** [comment]
@Dilbert66 I finally figured out my problem. Not sure if I looked at the incorrect data sheet and/or had a momentary short circuit in my brain when putting it all together. I had the transistor installed incorrectly. Quickly swapped it and all seems to be functioning correctly now.
